components/dashboard/src/start/CreateWorkspace.tsx

```diff
@@ -75,16 +76,10 @@ export default class CreateWorkspace extends React.Component<CreateWorkspaceProp
           </div>;
           break;
         case ErrorCodes.NOT_AUTHENTICATED:
-          const authorizeUrl = gitpodHostUrl.withApi({
-              pathname: '/authorize',
-              search: `returnTo=${encodeURIComponent(window.location.toString())}&host=${error.data.host}&scopes=${error.data.scopes.join(',')}`
-          }).toString();
-          window.location.href = authorizeUrl;
-          statusMessage = <div className="mt-2 flex flex-col space-y-8">
-            <p className="text-base w-96">Redirecting to authorize with {error.data.host} …</p>
-            <a href={authorizeUrl}><button className="secondary">Authorize with {error.data.host}</button></a>
-          </div>;
-          break;
+          const authorizeURL = `/api/authorize?returnTo=${encodeURIComponent(window.location.toString())}&host=${error.data.host}&scopes=${error.data.scopes.join(',')}`;
+          return <Redirect push to={authorizeURL}/>;
+        case ErrorCodes.USER_BLOCKED:
+          return <Redirect to="/blocked"/>;
```

Comment:
Same here 😕

<img width="1440" alt="Screenshot 2021-04-07 at 09 21 00" src="https://user-images.githubusercontent.com/599268/113826578-9e299500-9782-11eb-9e06-b4ca97072c17.png">

Reply:
That's because `<Redirect/>` only works for `<Route/>`s in the same `<Switch/>`.

- `/api/authorize` is not a route
- `/blocked` is currently outside the `<Switch/>`